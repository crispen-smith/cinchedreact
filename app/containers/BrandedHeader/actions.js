/*
 *
 * Brand actions
 *
 */

import BRAND_ACTIONS from './constants';

export function login() {
  return {
    type: BRAND_ACTIONS.LOGIN,
  };
}

export function logout() {
  return {
    type: BRAND_ACTIONS.LOGOUT,
  };
}
