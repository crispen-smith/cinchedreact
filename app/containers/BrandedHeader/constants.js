/*
 *
 * Header constants
 *
 */

import keymirror from 'keymirror';

const BRAND_ACTIONS = keymirror({
  LOGIN: null,
  LOGOUT: null,
});

export default BRAND_ACTIONS;
