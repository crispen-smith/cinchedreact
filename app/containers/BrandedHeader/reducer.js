import { fromJS } from 'immutable';
import BRAND_ACTIONS from './constants';
const initialState = fromJS({ loggedIn: true });

function brandReducer(state = initialState, action) {
  switch (action.type) {
    case BRAND_ACTIONS.LOGIN: {
      return fromJS({
        ...state.toJS(),
        loggedIn: true,
      });
    }
    case BRAND_ACTIONS.LOGOUT: {
      return fromJS({
        ...state.toJS(),
        loggedIn: false,
      });
    }

    default:
      return state;
  }
}

export default brandReducer;
