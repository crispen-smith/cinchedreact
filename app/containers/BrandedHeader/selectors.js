import { createSelector } from 'reselect';

/**
 * Direct selector to the brand state domain
 */
const selectBrandDomain = state => state.get('brand');

const makeSelectBrand = () =>
  createSelector(selectBrandDomain, substate => substate.toJS());

const makeSelectLoggedIn = () =>
  createSelector(selectBrandDomain, substate => substate.toJS().loggedIn);

export default makeSelectBrand;
export { selectBrandDomain, makeSelectLoggedIn };
