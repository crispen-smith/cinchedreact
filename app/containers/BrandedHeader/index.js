/**
 *
 * BrandedHeader
 *
 */

import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import injectSaga from 'utils/injectSaga';
import injectReducer from 'utils/injectReducer';

import { makeSelectLoggedIn } from './selectors';
import reducer from './reducer';
import saga from './saga';
import BrandBox from '../../components/BrandBox';
import LogToggle from '../../components/LogToggle';

import * as brandActions from './actions';
import { colors } from '../../resources/colors';

const BH = styled.div`
  display: flex;
  background-color: ${colors.pink};
  border-bottom: 1px solid rgba(255, 255, 255, 0.1);
  box-shadow: 1px 1px 1px 0px rgba(0, 0, 0, 0.1);
`;

class BrandedHeaderBase extends React.Component { // eslint-disable-line
  render() {
    const dispatch = { ...this.props };
    let action;
    if (this.props.loggedIn) {
      action = brandActions.logout();
    } else {
      action = brandActions.login();
    }

    const loggingSwitch = () => dispatch(action);
    return (
      <BH>
        <BrandBox />
        <LogToggle loggedIn={this.props.loggedIn} onClick={loggingSwitch} />
      </BH>
    );
  }
}

BrandedHeaderBase.propTypes = {
  loggedIn: PropTypes.bool.isRequired,
};

const mapStateToProps = createStructuredSelector({
  loggedIn: makeSelectLoggedIn(),
});

export function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'brand', reducer });
const withSaga = injectSaga({ key: 'brand', saga });

const BrandedHeader = compose(
  withReducer,
  withSaga,
  withConnect,
)(BrandedHeaderBase);

export default BrandedHeader;
