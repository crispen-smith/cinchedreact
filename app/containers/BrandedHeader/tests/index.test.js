import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import BrandedHeader from '../index';
import configureStore from '../../../configureStore';

describe('<BrandedHeader />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Renders a BrandBox', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrandedHeader />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
