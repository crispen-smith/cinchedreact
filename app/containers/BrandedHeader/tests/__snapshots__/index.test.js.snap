// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandedHeader /> Renders a BrandBox 1`] = `
<div
  className="BrandedHeader__BH-HxBPG cbfNhb"
>
  <div
    className="BrandBox__BrandBoxBase-bOpzwz dvPfFb"
  >
    <div
      className="MegaLogo__MegaLogoBase-gVmnHe cypxsg"
    >
      <span
        className="MegaLogo__C-hBLyQD CnhnX"
      >
        C
      </span>
      <span
        className="MegaLogo__C-hBLyQD iPmZWc"
      >
        T
      </span>
    </div>
  </div>
  <div />
</div>
`;
